index.go

```diff
@@ -158,32 +158,39 @@ func (i Index) GetStats() (resp *StatsIndex, err error) {
 	return resp, nil
 }
 
-func (i Index) GetTask(taskID int64) (resp *Task, err error) {
-	resp = &Task{}
-	req := internalRequest{
-		endpoint:            "/indexes/" + i.UID + "/tasks/" + strconv.FormatInt(taskID, 10),
-		method:              http.MethodGet,
-		withRequest:         nil,
-		withResponse:        resp,
-		acceptedStatusCodes: []int{http.StatusOK},
-		functionName:        "GetTask",
-	}
-	if err := i.client.executeRequest(req); err != nil {
-		return nil, err
-	}
-	return resp, nil
+func (i Index) GetTask(taskUID int64) (resp *Task, err error) {
+	return i.client.GetTask(taskUID)
 }
 
-func (i Index) GetTasks() (resp *ResultTask, err error) {
-	resp = &ResultTask{}
+func (i Index) GetTasks(param *TasksQuery) (resp *TaskResult, err error) {
+	resp = &TaskResult{}
 	req := internalRequest{
-		endpoint:            "/indexes/" + i.UID + "/tasks",
+		endpoint:            "/tasks?indexUid=" + i.UID,
```

Comment:
I think you don't need this `?indexUid=" + i.UID` since you will get the value from `withQueryParams["indexUid"]` correct?